Adding Hive detailed, tenant grafana dashboard

s-sabitova opened the pull request:
### Changelog entry <!-- a user-readable short description of the changes that goes to CHANGELOG.md and Release Notes -->

...

### Changelog category <!-- remove all except one -->

* Documentation (changelog entry is not required)

### Description for reviewers <!-- (optional) description for those who read this PR -->

Created a dashboard based on an internal one (Hive detailed). Only for tenant hives for now. Dashboard for the root hive will be added separately.
Detailed description and links to demo dashboards may be found in KIKIMR-22969.

s-sabitova committed:
Create hive-detailed-tenant.json

s-sabitova committed:
Merge branch 'ydb-platform:main' into hive-detailed-tenant-dashboard

s-sabitova committed:
Update and rename hive-detailed-tenant.json to database-hive-detailed.json

s-sabitova committed:
Update grafana-dashboards.md

s-sabitova committed:
Update grafana-dashboards.md

s-sabitova committed:
Update grafana-dashboards.md

vporyadke reviewed (dismissed):
>Created a dashboard based on an internal one (Hive detailed). Only for tenant hives for now. Dashboard for the root hive will be added separately.

Жесть, я думал, проблема, что нужно особым образом смотреть на метрки с корневого домена, актуальна только для старых внутренних кластеров, а оно и во вне так...

blinkov commented on ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -109,3 +109,18 @@ DataShard tablet metrics.
 
 Download the [datashard.json](https://raw.githubusercontent.com/ydb-platform/ydb/refs/heads/main/ydb/deploy/helm/ydb-prometheus/dashboards/datashard.json) file with the **DataShard** dashboard.
+
+## Database Hive {#database-hive-detailed}
+
+[Hive](../../../contributor/hive.md) metrics for the selected database.
+
+| Name | Description |
+|---|---|
+| hive+balancer | CPU time utilized by HIVE_ACTOR and HIVE_BALANCER_ACTOR |
+| hive event response time | How much time it takes for Hive to respond |
+| hive local transaction times | CPU time utilized by Hive when executing local transactions |
+| {{local-tx}} avg time | Count and average execution time of {{local-tx}} transactions. Transaction type {{local-tx}} is set in the filter on the dashboard |
+| tablet count | Total tablets count in the database |
+| event queue size | Database hive event queue size |
+| versions | Versions of YDB running on cluster nodes |
+| hive node | Node where the database hive is running |
```
```suggestion
## Database Hive {#database-hive-detailed}

[Hive](../../../contributor/hive.md) metrics for the selected database.

| Name | Description |
|---|---|
| hive+balancer | CPU time utilized by HIVE_ACTOR and HIVE_BALANCER_ACTOR |
| hive event response time | Time it takes for Hive to respond |
| hive local transaction times | CPU time utilized by Hive when executing local transactions |
| {{local-tx}} avg time | Count and average execution time of {{local-tx}} transactions. The transaction type {{local-tx}} is set in the filter on the dashboard. |
| tablet count | Total number of tablets in the database |
| event queue size | Size of the database Hive event queue |
| versions | Versions of {{ ydb-short-name }} running on cluster nodes |
| hive node | Node where the database Hive is running |
```

blinkov commented on ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -109,3 +109,18 @@ DataShard tablet metrics.
+| hive+balancer | CPU time utilized by HIVE_ACTOR and HIVE_BALANCER_ACTOR |
+| hive event response time | How much time it takes for Hive to respond |
+| hive local transaction times | CPU time utilized by Hive when executing local transactions |
+| {{local-tx}} avg time | Count and average execution time of {{local-tx}} transactions. Transaction type {{local-tx}} is set in the filter on the dashboard |
```
`{{ foo }}` is part of the Markdown syntax to use a variable value, and the `{{local-tx}}` variable doesn't seem to be defined, so you probably meant to just write this in text, which needs escaping.

blinkov replied:
Describe the dashboard filters separately and what they do.

ijon replied:
```suggestion
| `{{local-tx}}` avg time | Average execution time of a single local transaction for the type specified in the `local-tx` parameter selector on the dashboard. (`{{local-tx}}` appears when no transaction type is selected). |
```

blinkov commented on ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -109,3 +109,18 @@ DataShard tablet metrics.
+| hive+balancer | CPU time utilized by HIVE_ACTOR and HIVE_BALANCER_ACTOR |
+| hive event response time | How much time it takes for Hive to respond |
+| hive local transaction times | CPU time utilized by Hive when executing local transactions |
+| {{local-tx}} avg time | Count and average execution time of {{local-tx}} transactions. Transaction type {{local-tx}} is set in the filter on the dashboard |
```
List all possible `{{local-tx}}` values with some explanations.

s-sabitova replied:
There are 67 types of transactions - I'm not sure this article is the best place to list them. In my view, we could add the description of them (or the most important ones) to the hive documentation: https://github.com/ydb-platform/ydb/blob/main/ydb/docs/en/core/contributor/hive.md

blinkov replied:
Let's create a ticket/issue for that

blinkov commented on ydb/deploy/helm/ydb-prometheus/dashboards/database-hive-detailed.json:
```diff
@@ -0,0 +1,1130 @@
+{
```
There's a second place hosting the same dashboards  https://github.com/ydb-platform/ydb-ansible/blob/main/roles/monitoring_services/files/dashboards/ — let's try to keep them in sync. See YDBOPS-11070.

s-sabitova replied:
https://github.com/ydb-platform/ydb-ansible/pull/74

blinkov commented on ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -109,3 +109,18 @@ DataShard tablet metrics.
+| hive+balancer | CPU time utilized by HIVE_ACTOR and HIVE_BALANCER_ACTOR |
+| hive event response time | How much time it takes for Hive to respond |
+| hive local transaction times | CPU time utilized by Hive when executing local transactions |
+| {{local-tx}} avg time | Count and average execution time of {{local-tx}} transactions. Transaction type {{local-tx}} is set in the filter on the dashboard |
```
Describe what happens if nothing is selected on the filter.

s-sabitova replied:
This list is formed by fetching all possible transactions names from YDB counters metrics. There are no empty names, so it's not possible to select an empty value in the filter.

ijon commented on ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -109,3 +109,18 @@ DataShard tablet metrics.
+|---|---|
+| hive+balancer | CPU time utilized by HIVE_ACTOR and HIVE_BALANCER_ACTOR |
+| hive event response time | How much time it takes for Hive to respond |
+| hive local transaction times | CPU time utilized by Hive when executing local transactions |
```
```suggestion
| hive local transaction times | CPU time utilized by various local transaction types in Hive. Shows the structure of Hive load based on different activities. |
```

ijon commented on ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -109,3 +109,18 @@ DataShard tablet metrics.
+| hive local transaction times | CPU time utilized by Hive when executing local transactions |
+| {{local-tx}} avg time | Count and average execution time of {{local-tx}} transactions. Transaction type {{local-tx}} is set in the filter on the dashboard |
+| tablet count | Total tablets count in the database |
+| event queue size | Database hive event queue size |
```
```suggestion
| event queue size | Size of the incoming event queue in Hive. Consistently high values indicate Hive cannot process events fast enough. |
```

ijon commented on ydb/deploy/helm/ydb-prometheus/dashboards/database-hive-detailed.json:
```diff
@@ -0,0 +1,1130 @@
+          "refId": "A"
+        }
+      ],
+      "title": "hive event response time",
```
```suggestion
      "title": "hive self-ping time",
```

ijon reviewed (commented):
Add suggestions for graph descriptions

blinkov commented:
@s-sabitova, ping, do you intend to finish this?

s-sabitova committed:
Update ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

s-sabitova committed:
Merge branch 'ydb-platform:main' into hive-detailed-tenant-dashboard

s-sabitova committed:
Update grafana-dashboards.md

s-sabitova committed:
Update grafana-dashboards.md

s-sabitova committed:
Update grafana-dashboards.md

s-sabitova committed:
Update ydb/deploy/helm/ydb-prometheus/dashboards/database-hive-detailed.json

Co-authored-by: ijon <ivan.chelubeev@gmail.com>

s-sabitova committed:
Update grafana-dashboards.md

s-sabitova commented:
> @s-sabitova, ping, do you intend to finish this?

Made changes according to the comments

blinkov commented on ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -109,3 +109,23 @@ DataShard tablet metrics.
+
+| Name | Description |
+|---|---|
+| hive+balancer | CPU time utilized by HIVE_ACTOR and HIVE_BALANCER_ACTOR, the two of the most important actors of the Hive tablet. |
+| hive self-ping time | Time it takes Hive to respond to itself. High values indicate heavy load (and low responsiveness) of the Hive. |
+| hive local transaction times | CPU time utilized by various local transaction types in Hive. Shows the structure of Hive load based on different activities. |
+| `{{local-tx}}` avg time | Average execution time of a single local transaction for the type specified in the `local-tx` parameter selector on the dashboard. |
+| tablet count | Total number of tablets in the database. |
+| event queue size | Size of the incoming event queue in Hive. Consistently high values indicate Hive cannot process events fast enough. |
+| versions | Versions of {{ ydb-short-name }} running on cluster nodes. |
+| hive node | Node where the database Hive is running. |
```
```suggestion
| hive+balancer | CPU time utilized by `HIVE_ACTOR` and `HIVE_BALANCER_ACTOR`, two of the most important actors of the Hive tablet. |
| hive self-ping time | Time it takes Hive to respond to itself. High values indicate heavy load (and low responsiveness) of the Hive. |
| hive local transaction times | CPU time utilized by various local transaction types in Hive. Shows the structure of Hive load based on different activities. |
| `{{local-tx}}` avg time | Average execution time of a single local transaction for the type specified in the `local-tx` parameter selector on the dashboard. |
| tablet count | Total number of tablets in the database. |
| event queue size | Size of the incoming event queue in Hive. Consistently high values indicate Hive cannot process events fast enough. |
| versions | Versions of {{ ydb-short-name }} running on cluster nodes. |
| hive node | Node where the database Hive is running. |
```

blinkov commented on ydb/docs/ru/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -81,3 +81,23 @@
 | ReadSets | Сведения о пересылаемых ReadSets при исполнении пользовательской транзакции |
 | Other | Прочие метрики |
+
+## Database hive {#database-hive-detailed}
```
```suggestion
## Database Hive {#database-hive-detailed}
```

blinkov commented on ydb/docs/ru/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -81,3 +81,23 @@
+
+Метрики таблетки [Hive](../../../contributor/hive.md) выбранной базы данных.
+
+На дашборде размещены следующие фильтры:
+* database - используется для выбора базы данных, метрики которой необходимо отобразить;
+* ds - используется для выбора источника данных, подключенного к Grafana;
+* local-tx - определяет, какие данные будут отображаться в виджете `{{local-tx}}` avg time.
```
```suggestion
На дашборде размещены следующие фильтры:
* database — используется для выбора базы данных, метрики которой необходимо отобразить;
* ds — используется для выбора источника данных, подключённого к Grafana;
* local-tx — определяет, какие данные будут отображаться в виджете `{{local-tx}}` avg time.
```

blinkov commented on ydb/docs/ru/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -81,3 +81,23 @@
+
+| Имя | Описание |
+|---|---|
+| hive+balancer | Процессорное время, потребляемое HIVE_ACTOR и HIVE_BALANCER_ACTOR - двумя самыми важными акторами Hive. |
```
```suggestion
| hive+balancer | Процессорное время, потребляемое `HIVE_ACTOR` и `HIVE_BALANCER_ACTOR` — двумя самыми важными акторами Hive. |
```

s-sabitova committed:
Update ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

s-sabitova committed:
Update ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

s-sabitova committed:
Update ydb/docs/ru/core/reference/observability/metrics/grafana-dashboards.md

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

s-sabitova committed:
Update ydb/docs/ru/core/reference/observability/metrics/grafana-dashboards.md

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

s-sabitova committed:
Update ydb/docs/ru/core/reference/observability/metrics/grafana-dashboards.md

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

blinkov commented on ydb/docs/ru/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -81,3 +81,23 @@
+| `{{local-tx}}` avg time | Среднее время выполнения одной локальной транзакции типа, выбранного в фильтре local-tx. |
+| tablet count | Общее число таблеток в базе данных. |
+| event queue size | Размер очереди событий. |
+| versions | Версии YDB, запущенные на узлах кластера. |
```
```suggestion
| versions | Версии {{ ydb-short-name }}, запущенные на узлах кластера. |
```

blinkov committed:
Update ydb/docs/ru/core/reference/observability/metrics/grafana-dashboards.md

blinkov commented on ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -109,3 +109,23 @@ DataShard tablet metrics.
+| tablet count | Total number of tablets in the database. |
+| event queue size | Size of the incoming event queue in Hive. Consistently high values indicate Hive cannot process events fast enough. |
+| versions | Versions of {{ ydb-short-name }} running on cluster nodes. |
+| hive node | Node where the database Hive is running. |
```
```suggestion
| hive node | Node where the database Hive is running. |

Download the [datashard.json](https://raw.githubusercontent.com/ydb-platform/ydb/refs/heads/main/ydb/deploy/helm/ydb-prometheus/dashboards/database-hive-detailed.json) file with the **Database Hive** dashboard.
```

blinkov committed:
Update ydb/docs/en/core/reference/observability/metrics/grafana-dashboards.md

s-sabitova committed:
Update database-hive-detailed.json

Changed according to kobzonega's comments:
1. Panel names are now starting with capital letters
2. Removed unnecessary hive mentions from panel titles
3. hive+balancer -> CPU usage by HIVE_ACTOR, HIVE_BALANCER_ACTOR
4. Added legend where necessary
5. Added description for all panels
6. Aligned panels on the dashboard

s-sabitova committed:
Update database-hive-detailed.json

Changed variable name from local-tx to Tx type, so the name better speaks for itself

s-sabitova committed:
Update grafana-dashboards.md

s-sabitova committed:
Update grafana-dashboards.md

s-sabitova commented on ydb/docs/ru/core/reference/observability/metrics/grafana-dashboards.md:
```diff
@@ -81,3 +81,23 @@
+На дашборде размещены следующие фильтры:
+* database — используется для выбора базы данных, метрики которой необходимо отобразить;
+* ds — используется для выбора источника данных, подключённого к Grafana;
+* local-tx — определяет, какие данные будут отображаться в виджете `{{local-tx}}` avg time.
```
Название фильтра изменено на Tx type
Текст изменен на "Tx type — определяет тип транзакции, для которого будут выведены графики на панели `{{Tx type}}` average time"

s-sabitova committed:
Update grafana-dashboards.md

s-sabitova committed:
Update grafana-dashboards.md
